tests/_test_msui/test_msui.py

```diff
@@ -328,3 +333,101 @@ def test_flight_track_io(self, mockload, mocksave, mockq, mocki, mockw):
+        top_view1.cbChangeMapSection.setCurrentText("00 global (cyl)")
+        wms_settings1 = {
+            "url": "http://open-mss.org/",
+            "layer": "ecmwf_EUR_LL015.PLRelHum01",
+            "level": "200.0",
+            "styles": "",
+            "init_time": "2012-10-17T12:00:00Z",
+            "valid_time": "2012-10-17T12:00:00Z"
+        }
+        qtbot.wait(500)
+        top_view1.restore_wms_settings(wms_settings1)
+
+        window.create_new_flight_track()
+        assert window.listFlightTracks.count() == 2
+        flight_track2 = window.active_flight_track
+        assert flight_track2.name == "new flight track (2)"
+        assert "new flight track (2)" in window.activated_flight_tracks
+
+        waypoint_model2 = flight_track2
+        waypoint_model2.insertRows(0, 2, waypoints=[
+            ft.Waypoint(lat=22.44, lon=86.67, location="point1"),
+            ft.Waypoint(lat=67.77, lon=48.67, location="point2")
+        ])
+
+        # Open top view for second flight track
+        window.create_view("topview", flight_track2)
+        assert window.listViews.count() == 2
+        top_view2 = window.listViews.item(1).window
+        assert isinstance(top_view2, MSUITopViewWindow)
+        assert top_view2.view_type == "Top View"
+
+        top_view2.cbChangeMapSection.setCurrentText("00 global (cyl)")
+        wms_settings2 = {
+            "url": "http://open-mss.org/",
+            "layer": "ecmwf_EUR_LL015.PLW01",
+            "level": "250.0",
+            "styles": "",
+            "init_time": "2012-10-17T12:00:00Z",
+            "valid_time": "2012-10-17T12:00:00Z"
+        }
+        qtbot.wait(500)
+        top_view2.restore_wms_settings(wms_settings2)
+
+        assert "new flight track (1)" in window.flight_track_settings
+        settings1 = window.flight_track_settings["new flight track (1)"]
+        assert len(settings1["views"]) == 1
+        view_settings1 = settings1["views"][0]
+        assert view_settings1["view_type"] == "topview"
+        assert view_settings1["map_section"] == "00 global (cyl)"
+        assert view_settings1["wms"]["url"] == wms_settings1["url"]
+        assert view_settings1["wms"]["layer"] == wms_settings1["layer"]
+        assert view_settings1["wms"]["level"] == wms_settings1["level"]
+        assert view_settings1["wms"]["styles"] == wms_settings1["styles"]
+        assert view_settings1["wms"]["init_time"] == wms_settings1["init_time"]
+        assert view_settings1["wms"]["valid_time"] == wms_settings1["valid_time"]
+        assert settings1["global"]["flight_track_name"] == "new flight track (1)"
+        assert "mss_version" in settings1["global"]
+
+        with patch("PyQt5.QtWidgets.QMessageBox.warning", return_value=QtWidgets.QMessageBox.Yes):
+            window.close()
+
+        # Assert: Verify view_settings.json after closing
+        config_path = Path(constants.MSUI_CONFIG_PATH)
+        config_path.mkdir(parents=True, exist_ok=True)
```

Comment:
This shouldn't be necessary, after all the directory has to exist for there to be the view_settings.json file in it, otherwise the first assert will already fail.